### PR TITLE
The wiki of the offset API has been updated again

### PR DESCRIPTION
**Pull request by eapache**
This should bring us up to date.

@Shopify/kafka 

Somebody should verify against https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-OffsetCommitRequest

### CODE REVIEW

File: offset_commit_request.go [hunk 1]
```diff
@@ -13,44 +13,71 @@ type offsetCommitRequestBlock struct {
...
+
 	return pe.putString(r.metadata)
 }
 
 type OffsetCommitRequest struct {
-	ConsumerGroup string
-	Version       int16 // 0 (0.8.1 and later) or 1 (0.8.2 and later, includes timestamp field)
-	blocks        map[string]map[int32]*offsetCommitRequestBlock
+	ConsumerGroup           string
+	ConsumerGroupGeneration int32  // v1 or later
```
Comment (wvanbergen) [inline offset_commit_request.go]: Do you have any idea what this used for? Couldn't find anything useful in the wiki.

Comment (eapache) [inline offset_commit_request.go]: https://cwiki.apache.org/confluence/display/KAFKA/Kafka+0.9+Consumer+Rewrite+Design

File: offset_fetch_request.go [hunk 2]
```diff
@@ -2,10 +2,15 @@ package sarama
 
 type OffsetFetchRequest struct {
 	ConsumerGroup string
+	Version       int16
 	partitions    map[string][]int32
 }
 
 func (r *OffsetFetchRequest) encode(pe packetEncoder) (err error) {
+	if r.Version < 0 || r.Version > 1 {
```
Comment (yagnik) [inline offset_fetch_request.go]: why not r.version != 1, you don't have decimal version do you ?

Comment (yagnik) [inline offset_fetch_request.go]: actually it should be > 2

**Comment by yagnik** [comment]
minor comment other lgtm

File: offset_fetch_request.go (hunk 2 above)
Comment (eapache) [inline offset_fetch_request.go]: I'm not sure what you mean? Based on my reading of the protocol guide (and [Jun's comment in JIRA](https://issues.apache.org/jira/browse/KAFKA-1841?focusedCommentId=14270222&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14270222)) the version can be 0 or 1, so if it is `<0` or `>1` we need to bail.

**Comment by wvanbergen** [comment]
This LGTM.

Not sure how long the approach of trying to support all version from one type is going to be sustainable, but for now :shipit:.

File: offset_fetch_request.go (hunk 2 above)
Comment (yagnik) [inline offset_fetch_request.go]: Nevermind then, that looks good

**Comment by jsvisa** [comment]
@eapache @wvanbergen  Seems Kafka 0.8.2 also not support `OffsetManage`. Before you send a `OffsetCommitRequest`, you need to send a `ConsumerMetadataRequest` to get the `Coordinator`, but this request always failed in kafka 0.8.2.0:

```bash
kafka server: Offset's topic has not yet been created. CoordinatorID:-1 CoordinatorHost: CoordinatorPort:-1
```
which is succeed in Kafka 0.8.2-beta.

**Comment by wvanbergen** [comment]
@jsvisa It is working for me on 0.8.2.1 (see #379). The very first time you make this API call, Kafka first has to create the topic which takes some time. The only solution that works is retrying often enough.

**Comment by jsvisa** [comment]
@wvanbergen  Thanks for your advise, I'll try it later in 0.8.2.1

**Pull request merged**